docker/dispatch.sh

```diff
@@ -1,13 +1,69 @@
 #!/usr/bin/env sh
 
+_term() {
+  docker kill "$child"
+  status=$(docker inspect "$child" --format='{{.State.ExitCode}}')
+  docker rm "$child"
+  exit "$status"
+}
+
 case "$(uname)" in
 	Darwin*)	HOST_MACHINE=Mac;;
 	*)		HOST_MACHINE=Linux;;
 esac
 
+volume_mounts=""
+
+# Get the container's CARGO_HOME ENV, defaulting to /usr/local/cargo
+cargo_home=$(docker inspect -f \
+        '{{range $index, $value := .Config.Env}}{{println $value}}{{end}}' \
+            "$3" | grep CARGO_HOME | cut -d"=" -f 2)
+if [ -z "$cargo_home" ]; then
+	cargo_home="/usr/local/cargo"
+fi
+
+[ ! -d "$CARGO_CACHE" ] && mkdir -p "$CARGO_CACHE"
+
+volume_mounts="$volume_mounts -v $CARGO_CACHE:$cargo_home/registry:Z"
```

Comment:
Will `$volume_mounts` ever not be just `""` because of line 15?

Reply:
Yeh, it's quite possible, in that case this line should resolve to " -v $CARGO_CACHE:$cargo_home/registry:Z", which is still valid